mltsp/build_model.py

```diff
@@ -31,21 +32,71 @@ def rectangularize_featureset(featureset):
     return feature_df.loc[featureset.name]  # preserve original row ordering
 
 
+def fit_model_optimize_hyperparams(data, targets, model, model_params,
+                                   params_to_optimize):
+    """Optimize estimator hyperparameters.
+
+    Perform hyperparamter optimization using
+    `sklearn.grid_search.GridSearchCV`.
+
+    Parameters
+    ----------
+    data : Pandas.DataFrame
+        Features for training model.
+    targets : Pandas.Series
+        Targets corresponding to feature vectors in `data`.
+    model : sklearn estimator object
+        The model/estimator whose hyperparameters are to be optimized.
+    model_params : dict or list of dict
+        Dictionary with parameter names as keys and lists of parameter values
+        to try as values, or a list of such dictionaries.
+    params_to_optimize : list of str
+        List of parameter names to be optimized.
+
+    Returns
+    -------
+    `sklearn.grid_search.GridSearchCV` estimator object
+
+    """
+    # To fit with fixed, non-optimized params, must be wrapped in list
+    if isinstance(model_params, dict):
```

Comment:
if isinstance(model_params, dict):
        model_params = [model_params]
    for i in range(len(model_params)):
        ...